### ISSUE TITLE
PutObjectCommand via AWS JS SDK does not fire bucket notifications

### ISSUE THREAD

**Issue by simoncollins** [localstack/localstack#3732]
# Type of request: This is a ...

[X] bug report
[ ] feature request

# Detailed description

When I upload an object using the `PutObjectCommand` using the JS SDK and I have bucket notifications enabled, it does not trigger any notifications.

I have debugged this and it seems that the presence of the AWS X-ray ID (`x-id=PutObject`) in the request querystring is preventing `is_query_allowable` in `s3_listener.py` from returning `true` for this request which means that `should_send_notifications` is set to `false` even though the query is one of the correct types and meets the other criteria in the `all` check behind that flag.

## Expected behavior

Bucket notifications are correctly fired when applied to the bucket.

## Actual behavior

Bucket notifications are not attempted because localstack thinks the request does not meet the requirements.

# Steps to reproduce

## Command used to start LocalStack
`docker-compose up` given:

```yaml
version: '3'
services:
  aws-localstack:
   # localstack:0.12.6
    image: localstack/localstack:0.0@sha256:7ea76feaa8538a9cbd54cc25fc9157b3c1fd6acbdce9ee6873a0d88a07585dd5
    ports:
      - '4566:4566'
    environment:
      - SERVICES=s3,iam,lambda,cloudwatch
      - DEFAULT_REGION=ap-southeast-2
      - LAMBDA_EXECUTOR=docker
      - DEBUG=1
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

```javascript
await s3.send(new PutObjectCommand({
            Bucket: 'some-bucket',
            Key: "index.html",
            Body: "<body>hello world!</body>"
        }))
```

**Comment by macnev2013** [comment]
Thanks for reporting @simoncollins.
Is it possible to provide a small isolated example with deployment commands (a git repo) to reproduce the issue?
Thanks

**Comment by simoncollins** [comment]
Sure, I'll put something together.

**Comment by macnev2013** [comment]
Thanks @simoncollins.

**Comment by whummer** [comment]
Thanks @simoncollins @macnev2013 . This should be fixed in #3799 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks!

**Comment by simoncollins** [comment]
Thanks @whummer ... my code is still not working with the latest image but it may well be something wrong I'm doing so I'll have to debug further.
